src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let linkedStringParts = '';
+    let currentElement;
+
+    for (let i = 0; i < this.length; i++) {
+      currentElement = this[i];
+
+      if (currentElement === null || currentElement === undefined) {
+        currentElement = '';
```

Comment:
maybe set '' as a default value and just continue the loop here instead of reassigning it